packages/tuono/src/build/index.ts

```diff
@@ -3,86 +3,102 @@ import { build, createServer, mergeConfig } from 'vite'
 import { LazyLoadingPlugin } from 'tuono-lazy-fn-vite-plugin'
-import mdx from '@mdx-js/rollup'
+
+import type { TuonoConfig } from '../config'
 
 import { loadConfig, blockingAsync } from './utils'
 
 const VITE_PORT = 3001
 
-const BASE_CONFIG: InlineConfig = {
-  root: '.tuono',
-  logLevel: 'silent',
-  publicDir: '../public',
-  cacheDir: 'cache',
-  envDir: '../',
-  optimizeDeps: {
-    exclude: ['@mdx-js/react'],
-  },
-  plugins: [
-    { enforce: 'pre', ...mdx({ providerImportSource: '@mdx-js/react' }) },
-    // @ts-expect-error: TS configuration issue.
-    react({ include: /\.(jsx|js|mdx|md|tsx|ts)$/ }),
-    ViteFsRouter(),
-    LazyLoadingPlugin(),
-  ],
+/**
+ * From a given {@link TuonoConfig} return a `vite` "mergeable" {@link InlineConfig}
+ * including all default tuono related options
+ */
+function createBaseViteConfigFromTuonoConfig(
+  tuonoConfig: TuonoConfig,
+): InlineConfig {
+  const viteBaseConfig: InlineConfig = {
+    root: '.tuono',
+    logLevel: 'silent',
+    publicDir: '../public',
+    cacheDir: 'cache',
+    envDir: '../',
+
+    resolve: {
+      alias: tuonoConfig.vite?.alias ?? {},
+    },
+
+    optimizeDeps: tuonoConfig.vite?.optimizeDeps,
+
+    plugins: [
+      ...(tuonoConfig.vite?.plugins ?? []),
+      react(),
+      ViteFsRouter(),
+      LazyLoadingPlugin(),
+    ],
+  }
+
+  // seems redundant but it's useful to log the value when debugging, until we have a logging infrastructure.
+  return viteBaseConfig
 }
 
 const developmentSSRBundle = (): void => {
   blockingAsync(async () => {
     const config = await loadConfig()
     await build(
-      mergeConfig(BASE_CONFIG, {
-        resolve: {
-          alias: config.vite?.alias || {},
-        },
-        build: {
-          ssr: true,
-          minify: false,
-          outDir: 'server',
-          emptyOutDir: true,
-          rollupOptions: {
-            input: './.tuono/server-main.tsx',
-            onLog() {
-              // Silence all logs
-            },
-            output: {
-              entryFileNames: 'dev-server.js',
-              format: 'iife',
+      mergeConfig<InlineConfig, InlineConfig>(
+        createBaseViteConfigFromTuonoConfig(config),
+        {
+          build: {
+            ssr: true,
+            minify: false,
+            outDir: 'server',
+            emptyOutDir: true,
+            rollupOptions: {
+              input: './.tuono/server-main.tsx',
+              onLog() {
+                /* Silence all logs */
+              },
+              output: {
+                entryFileNames: 'dev-server.js',
+                format: 'iife',
+              },
             },
           },
+          ssr: {
+            target: 'webworker',
+            noExternal: true,
+          },
         },
-        ssr: {
-          target: 'webworker',
-          noExternal: true,
-        },
-      }),
+      ),
     )
   })
 }
 
 const developmentCSRWatch = (): void => {
+  console.info('developmentCSRWatch')
```

Comment:
I'd remove this to leave all the logs on rust's side.

Reply:
An oversight when I removed the debug logs 😩